STYLEGUIDE.md

```diff
@@ -226,22 +226,6 @@ const keys = ['foo', 'bar'],
       values = [23, 42];
 ```
 
-### Prefix private class methods with an underscore `JS only`
```

Comment:
As mentioned by @stacey-gammon [here](https://github.com/elastic/kibana/pull/46464#discussion_r327767876):

> I would get rid of this one, or even make it the opposite recommendation. A) it's js only so not really relevant anymore since everyone should be writing in typescript. B) even if it is js, it should eventually be ts, at which point we'll have to rewrite it all to not have underscores.

I totally agree with that, it makes actually refactoring to TypeScript later easier, and we've seen a couple of cases we actually used underscored method names, and they are still used all around the place.